homeassistant/components/tailscale/config_flow.py

```diff
@@ -0,0 +1,122 @@
+"""Config flow to configure the Tailscale integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from tailscale import Tailscale, TailscaleAuthenticationError, TailscaleError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_TAILNET, DOMAIN
+
+
+async def validate_input(hass: HomeAssistant, *, tailnet: str, api_key: str) -> None:
+    """Try using the give tailnet & api key against the Tailscale API."""
+    session = async_get_clientsession(hass)
+    tailscale = Tailscale(
+        session=session,
+        api_key=api_key,
+        tailnet=tailnet,
+    )
+    await tailscale.devices()
+
+
+class TailscaleFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow for Tailscale."""
+
+    VERSION = 1
+
+    reauth_entry: ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                await validate_input(
+                    self.hass,
+                    tailnet=user_input[CONF_TAILNET],
+                    api_key=user_input[CONF_API_KEY],
+                )
+            except TailscaleAuthenticationError:
+                errors["base"] = "invalid_auth"
+            except TailscaleError:
+                errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(user_input[CONF_TAILNET])
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=user_input[CONF_TAILNET],
+                    data={
+                        CONF_TAILNET: user_input[CONF_TAILNET],
+                        CONF_API_KEY: user_input[CONF_API_KEY],
+                    },
+                )
+        else:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_TAILNET, default=user_input.get(CONF_TAILNET)
```

Comment:
Maybe default to empty string to be in line with the validator type. Since the item is required it will require the user to enter something still.
```suggestion
                        CONF_TAILNET, default=user_input.get(CONF_TAILNET, "")
```